.pre-commit-hooks.yaml

```diff
@@ -22,3 +22,12 @@
   pass_filenames: false
   files: ^(.*/)?poetry\.lock$
   args: ["-f", "requirements.txt", "-o", "requirements.txt"]
+
+- id: poetry-update
+  name: poetry-update
+  description: Update the Poetry dependencies if out of sync
+  entry: poetry install --sync
```

Comment:
The `--sync` argument should be moved to the [args](https://pre-commit.com/#hooks-args) key. Doing so users can decide to overwrite it.

I wonder if want to use `--sync` by default at all. I personally would expect this, but having a look into responses in our issue tracker, there seems to be a lot of people who don't agree. @radoering, @Secrus, @neersighted Any opinions?

Reply:
Yeah, I would remove it. Personally, I rarely use `--sync` option.

Reply:
So, if I remove some dependencies and invoke `poetry install` does it result in the same virtual environment as invoking `poetry install --sync`? If so, what is the correct use case for the `--sync` option?

Reply:
`poetry install` without `--sync` will not remove any dependencies. This is useful for people who have installed additional things in the venv via `pip` and want to keep these things. Personally, I clearly prefer using `--sync`. :smiley:

Reply:
I would like to put the `--sync` option into the args key, because for me it is the intended behaviour. If I change branches and there is a different lock file, i want to have a clean  slate

Reply:
I always use `--sync` so that's fine for me. 😉

We could also add two hooks `poetry-install` (without sync) and `poetry-sync` (with sync). Maybe, we just add `poetry-sync` now and poetry-install later only if someone requests it?

Reply:
To be consistent with the existing pre-commit hooks, I think it would be a good idea to call this hook `poetry-install` (because it runs `poetry install`). One can add any argument they want  (in you use case `--sync`) to projects pre-commit config then.

Reply:
That's fine with me, too. It has the benefit that it's more flexible and we only need one hook. 🤷